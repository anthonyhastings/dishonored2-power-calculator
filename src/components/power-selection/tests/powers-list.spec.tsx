import renderer from 'react-test-renderer';
import type { PowersListProps } from '../powers-list';
import PowersList from '../powers-list';

describe('PowersList component', () => {
  let testContext: {
    component?: renderer.ReactTestRenderer;
    defaultProps: PowersListProps;
  };

  const renderComponent = (props = {}) => (
    <PowersList {...testContext.defaultProps} {...props} />
  );

  beforeEach(() => {
    testContext = {
      defaultProps: {
        children: 'Test Title',
        powers: [
          {
            id: 'enhancement-id',
            parentPowerId: null,
            characterId: 'character-id',
            type: 'enhancement',
            name: 'Top-level Enhancement',
            description: 'Description for a top-level enhancement',
            cost: 1,
          },
        ],
      },
    };
  });

  describe('when rendered', () => {
    beforeEach(() => {
      testContext.component = renderer.create(renderComponent());
    });

    it('renders a title and grid of powers', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when className is provided', () => {
    it('renders with extra className', () => {
      expect(renderer.create(renderComponent())).toMatchDiffSnapshot(
        renderer.create(
          renderComponent({
            className: 'mock-class',
          })
        )
      );
    });
  });
});
