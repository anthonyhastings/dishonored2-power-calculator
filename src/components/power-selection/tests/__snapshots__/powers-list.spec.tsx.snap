// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowersList component when className is provided renders with extra className 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,7 +1,7 @@
  <section
-   className="powers-list"
+   className="powers-list mock-class"
  >
    <h1
      className="powers-list__title"
    >
      Test Title
`;

exports[`PowersList component when rendered renders a title and grid of powers 1`] = `
<section
  className="powers-list"
>
  <h1
    className="powers-list__title"
  >
    Test Title
  </h1>
  <div
    className="powers-list__grid"
  >
    <div
      className="powers-list__grid-item"
    >
      <h1>
        Top-level Enhancement
      </h1>
      <span
        aria-label="Symbol for Top-level Enhancement"
        className="powers-list__icon powers-list__icon--top-level-enhancement"
      />
    </div>
  </div>
</section>
`;
